### PR TITLE
Add codeready user readonly theme

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### Reference issue
https://issues.redhat.com/browse/CRW-1379

Screenshot from 2020-11-18 18-17-03

Screenshot from 2020-11-18 18-18-16

### CODE REVIEW

**Review by AndrienkoAleksandr — APPROVED** [review]

File: pkg/deploy/identity-provider/deployment_keycloak.go [hunk 1]
```diff
@@ -489,13 +490,18 @@ func getSpecKeycloakDeployment(
 		" && /opt/jboss/docker-entrypoint.sh -b 0.0.0.0 -c standalone.xml"
 	command += " -Dkeycloak.profile.feature.token_exchange=enabled -Dkeycloak.profile.feature.admin_fine_grained_authz=enabled"
 	if cheFlavor == "codeready" {
+		addUsernameReadonlyTheme := "if [ ! -d /opt/eap/themes/codeready-username-readonly ]; then" +
```
Comment (nickboldt) [inline pkg/deploy/identity-provider/deployment_keycloak.go:493]: I love that this project allows 'if product then...' branches -- makes the sync/fork/massaging easier downstream :)

File: pkg/deploy/identity-provider/deployment_keycloak.go [hunk 2]
```diff
@@ -12,12 +12,13 @@
...
 
 import (
-	"github.com/eclipse/che-operator/pkg/deploy/server"
 	"context"
 	"regexp"
 	"strconv"
 	"strings"
 
+	"github.com/eclipse/che-operator/pkg/deploy/server"
+
```
Comment (nickboldt) [inline pkg/deploy/identity-provider/deployment_keycloak.go:21]: strange spacing and sort order, but I'm not fussed so +1. :D

**Comment by nickboldt** [comment]
LGTM but not an expert in the code. +1

File: pkg/deploy/identity-provider/deployment_keycloak.go [hunk 3]
```diff
@@ -489,13 +490,20 @@ func getSpecKeycloakDeployment(
...
 	command += " -Dkeycloak.profile.feature.token_exchange=enabled -Dkeycloak.profile.feature.admin_fine_grained_authz=enabled"
 	if cheFlavor == "codeready" {
+		addUsernameReadonlyTheme := "baseTemplate=/opt/eap/themes/base/login/login-update-profile.ftl" +
+			" && readOnlyTemplateDir=/opt/eap/themes/codeready-username-readonly/login" +
+			" && readOnlyTemplate=${readOnlyTemplateDir}/login-update-profile.ftl" +
+			" && if [ ! -d ${readOnlyTemplateDir} ]; then" +
+			" mkdir -p ${readOnlyTemplateDir}" +
+			" && cp ${baseTemplate} ${readOnlyTemplate}" +
+			" && echo \"parent=rh-sso\" > ${readOnlyTemplateDir}/theme.properties" +
+			" && sed -i 's|id=\"username\" name=\"username\"|id=\"username\" readonly name=\"username\"|g' ${readOnlyTemplate}; fi"
```
Comment (ibuziuk) [inline pkg/deploy/identity-provider/deployment_keycloak.go:500]: @tolusha could you please calrify when / on which stage this command is executed?
I'm wondering if it is possible to do such a manipulation manually for CRW 2.5

Comment (tolusha) [reply pkg/deploy/identity-provider/deployment_keycloak.go:500]: The command is run inside keycloak pod before keycloak starts up.
I don't see a way how we can do it manually.

**Pull request merged**